packages/eslint-plugin-baseui/__tests__/deprecated-theme-api-test.js

```diff
@@ -0,0 +1,256 @@
+        const Foo = styled("div", ({ $theme }) => {
+          return {
+            color: $theme.colors.shadowFocus
+          };
+        });
+      `,
+      errors: [{messageId: MESSAGES.deprecateThemeProperty.id}],
+    },
+
+    // Invalid use of a deprecated theme property that is partially destructured via withStyle
+    {
+      code: `
+        const Foo = withStyle(Foo, ({ $theme }) => {
+          return {
+            backgroundColor: $theme.colors.shadowFocus
+          };
+        });
+      `,
+      errors: [{messageId: MESSAGES.deprecateThemeProperty.id}],
+    },
+
+    // Invalid use of a deprecated theme property that is partially destructured via overrides
+    {
+      code: `
+        function Foo() {
+          return (
+            <Button
+              overrides={{
+                Root: {
+                  style: ({ $theme }) => {
+                    return {
+                      color: $theme.colors.shadowFocus
+                    };
+                  }
+                }
+              }}
+            >
+              Foo
+            </Button>
+          );
+        }
+      `,
+      errors: [{messageId: MESSAGES.deprecateThemeProperty.id}],
+    },
+
+    // Invalid use of a deprecated theme property that is partially destructured via styled
+    {
+      code: `
+        const Foo = styled("div", ({ $theme }) => {
+          const [colors] = $theme;
+          return {
+            color: colors.shadowFocus
+          };
+        });
+      `,
+      errors: [{messageId: 'deprecateThemeProperty'}],
+    },
+    {
+      code: `
+        const Foo = styled("div", ({$theme: {colors: {shadowFocus}}}) => ({
+          color: shadowFocus
+        }));
+      `,
+      errors: [
+        {messageId: MESSAGES.deprecateThemeProperty.id},
+        {messageId: MESSAGES.deprecateThemeProperty.id},
+      ],
+    },
+
+    // Invalid use of a deprecated theme property that is fully destructured via useStyletron
+    {
+      code: `
+        function Foo() {
+          const [
+            css,
+            {
+              colors: { shadowFocus }
+            }
+          ] = useStyletron();
+          return (
+            <div
+              className={css({
+                color: shadowFocus
+              })}
+            >
+              Foo
+            </div>
+          );
+        }
+      `,
+      errors: [
+        {messageId: MESSAGES.deprecateThemeProperty.id},
+        {messageId: MESSAGES.deprecateThemeProperty.id},
+      ],
+    },
+
+    // Invalid use of old typography values
+    {
+      code: `
+        const Foo = styled("div", ({ $theme }) => ({
+          ...$theme.typography.font300
+        }));
+      `,
+      errors: [{messageId: MESSAGES.replaceThemeProperty.id}],
+      output: `
+        const Foo = styled("div", ({ $theme }) => ({
+          ...$theme.typography.ParagraphMedium
+        }));
+      `,
+    },
+
+    // Invalid setting deprecated property in createTheme
+    {
```

Comment:
does the code below pass or fail? (create theme not imported from baseui)

```
function createTheme() {}
createTheme({colors: {shadowFocus: 'pink'}});
```

Reply:
Currently `shadowFocus` will be flagged.

Reply:
I'm going to add a couple test cases for this and add a check for `createTheme` being imported.

Reply:
Fixed in refactor